zyte_api/aio/retry.py

```diff
@@ -51,7 +51,7 @@ def _is_network_error(exc: BaseException) -> bool:
 
 
 def _is_throttling_error(exc: BaseException) -> bool:
-    return isinstance(exc, RequestError) and exc.status in (429, 503)
+    return isinstance(exc, RequestError) and exc.status in (429, 503, 520)
```

Comment:
520 is not a throttling error, I think it's a real error

Reply:
It might require different retry policy, and also we may want to have it configurable

Reply:
It is described as a temporary download error in the docs, but I know of at least 1 URL that consistently gets this error, so yes, retrying forever is probably not a good option.

Reply:
FTR, zyte-autoextract handled a similar case with [max_query_error_retries](https://github.com/zytedata/zyte-autoextract/blob/696421817eefca8720395249551c85fd4c838238/autoextract/aio/client.py#L164) option.